doc/mgmt/SONiC_AliasFeature_HLD.md

```diff
@@ -0,0 +1,337 @@
+# Alias Feature
+
+Implement alternative naming convention (alias) for ethernet interfaces via  CLI/REST/gNMI  in SONiC management framework.
+
+# High Level Design Document
+
+#### Rev 0.1
+
+# Table of Contents
+
+- [List of Tables](#list-of-tables)
+- [Revision](#revision)
+- [About This Manual](#about-this-manual)
+- [Scope](#scope)
+- [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+
+| Rev  | Date       | Author       | Change Description |
+| ---- | ---------- | ------------ | ------------------ |
+| 0.1  | 05/01/2020 | Justine Jose | Initial draft      |
+|      |            |              |                    |
+|      |            |              |                    |
+
+# About this Manual
+
+This document provides general information about alias related syntax & overview for ethernet Interfaces in break out & non-breakout mode.
+
+# Scope
+
+Covers general design for supporting alias feature.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+# 1 Feature Overview
+
+The document covers the various interfaces for alias feature using SONiC management framework. Currently in SONiC, the ethernet interfaces are shown as a flat number. This flat interface numbering does not give mapping about actual ports that are mapped to the interface number. By enabling alias feature, user has better usability & mapping for actual port numbers.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide ability to have an alternative configuration for Ethernet Interface using SONiC management framework.
+
+### 1.1.2 Configuration and Management Requirements
+
+- Two modes to represent interfaces "native mode" & "alias mode".
+- Implement alias related config & show commands.
+- REST & gNMI set/get support for alias mode.
+
+### 1.1.3 Scalability Requirements
+
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+
+- By default system will boot in native mode. Native mode is where interfaces are displayed as Ethernet 1, Ethernet 4, Ethernet 8 etc. This also ensures that backwards compatibility is maintained.
+- Once alias is enabled, all subsequent CLI, REST & gNMI interfaces will use the alternative name for ethernet interface configuration & retrieval.
+- Existing SSH sessions : TBD.  
+
+### 1.2.2 Container
+
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+
+## 2.2 Functional Description
+
+# 3 Design
+
+## 3.1 Overview
+
+- When alias mode is enabled, ethernet interfaces will be represented as Eth[slot/port/breakout-port], where slot will start from 1 and port/breakout-port numbers will start from 1. This name format is fixed and will not have references to interface speed. For e.g Eth1/1 (in non breakout mode), Eth1/1/1 (in breakout mode).

```

Comment:
Any specific reason for starting the port from 1, and not from 0. 
Also breakout mode. I think Eth 1/1:1 is more indicative and clear, instead of 1/1/1

Reply:
The format & port numbering agreed by both teams earlier !